conda_build/main_develop.py

```diff
@@ -42,6 +43,21 @@ def main():
                    help=("Relink compiled extension dependencies against "
                          "libraries found in current conda env. "
                          "Do not add source to conda.pth."))
+    p.add_argument(
+                   '-b', '--build_ext',
+                   action='store_true',
+                   help=("Build extensions inplace, invoking: "
+                         "python setup.py build_ext --inplace; "
+                         "add to conda.pth; relink runtime libraries to "
+                         "environment's lib/."))
+    p.add_argument(
+                   '-c', '--clean_build_ext',
```

Comment:
Here's the problem. Currently, the logic for develop is as follows:
$ conda develop <options> path

if build_ext is in options: then invoke build
if np_pth_file is in options: then skip adding the path to conda.pth

At the end of logic based on input options, look for *.so in path and ensure they use conda environment for loading libraries at runtime.

If user tries any of the following:
$ conda develop path
$ conda develop --clean --build_ext path
$ conda develop --build_ext path

It's fine to add path to conda.pth unless --no-pth-file is given, and relink shared objects in path. However, if user does:
$ conda develop --clean path

it should only invoke $ python setup.py clean. It is seems unclean to add a path to conda.pth or try to relink shared objects when clean is invoked. I don't think the default behavior of conda develop that adds to conda.pth and relinks shared objects should not apply for --clean by itself. It doesn't break anything but doesn't seem right to do anything other than clean when conda develop --clean path is given.